experiment_handler/experiment_handler.py

```diff
@@ -159,45 +172,54 @@ def experiment_handler(semaphore):
 
     Experiment = retrieve_experiment()
     experiment_update = False
-    for method_name, obj in inspect.getmembers(Experiment, inspect.ismethod):
+    for method_name, obj in inspect.getmembers(Experiment, inspect.isfunction):
         if method_name == 'update':
             experiment_update = True
+
     if __debug__:
-        print("Experiment has update method: " + str(experiment_update))
+        if experiment_update:
+            printing("Experiment has an updated method.")
 
     exp = Experiment()
-    change_flag = False
+    global change_flag
```

Comment:
hmm. I was playing around with this when it wasn't working how I thought it should, but I don't think this is needed here because update_experiment should have access to change_flag since both are defined within experiment_handler